src/events/dns.js

```diff
@@ -0,0 +1,358 @@
+ */
+const initialDnsEvent = (name) => {
+    const startTime = Date.now();
+    const resource = new serverlessEvent.Resource([
+        'dns',
+        'dns',
+        name,
+    ]);
+    const dnsEvent = new serverlessEvent.Event([
+        `dns-${uuid4()}`,
+        utils.createTimestampFromTime(startTime),
+        null,
+        'dns',
+        0,
+        errorCode.ErrorCode.OK,
+    ]);
+    dnsEvent.setResource(resource);
+    return { dnsEvent, startTime };
+};
+
+/**
+ * Adding callback data/error to event, and finalize event.
+ * @param {serverlessEvent.Event} dnsEvent Dns event.
+ * @param {number} startTime Event start time.
+ * @param {Error} err Callback error.
+ * @param {string[] | Object[] | Object} metadata Callback metadata.
+ */
+const finalizeEvent = (dnsEvent, startTime, err, metadata) => {
+    if (err) {
+        eventInterface.setException(dnsEvent, err);
+    } else {
+        eventInterface.addToMetadata(dnsEvent, metadata);
+    }
+    dnsEvent.setDuration(utils.createDurationTimestamp(startTime));
+};
+
+/**
+ * Checking resource record type by resolve function name (default resolve4 - 'A').
+ * @param {*} arg1 function first argument.
+ * @param {*} arg2 function second argument
+ * @param {*} arg3 function third argument
+ * @param {*} functionName function resolve name.
+ * @returns {Object} Object of Hostname to resolve, Resource record type, and callback.
+ */
+const getRrtypeArguments = (arg1, arg2, arg3, functionName) => {
+    const hostname = arg1;
+    let rrtype = arg2;
+    let callback = arg3;
+    // in case of arg3 doesnt exist or arg2 is options.
+    if (!arg3 || typeof arg2 === 'object') {
+        if (functionName) {
+            rrtype = Object.values(rrtypesMethods).find(type => functionName.toLocaleLowerCase().includes((`query${type}`.toLocaleLowerCase())));
+        }
+        if (!rrtype) {
+            rrtype = rrtypesMethods.resolve4;
+        }
+        if (!arg3) {
+            callback = arg2;
+        }
+    }
+    return { hostname, rrtype, callback };
+};
+
+/**
+ * Gettign lookup arguments with options.
+ * @param {*} arg1 function first argument.
+ * @param {*} arg2 function second argument
+ * @param {*} arg3 function third argument
+ * @returns {Object} Object of Hostname, options (if doesnt exist- undifiend) and callback.
+ */
+const getLookupArguments = (arg1, arg2, arg3) => {
+    const hostname = arg1;
+    let options = arg2;
+    let callback = arg3;
+    if (!arg3) {
+        options = undefined;
+        callback = arg2;
+    }
+    return { hostname, options, callback };
+};
+
+/**
+ * Gettinng callback object of dns resolve.
+ * @param {*} arg callback argument
+ * @param {string} rrtype resouece record type
+ * @returns {Object} Returns the callback object with the appropriate key compared to rrtype,
+ * (if doesnt exist - records).
+ */
+const getCallbackResolveArgument = (arg, rrtype) => {
+    if (!arg) return undefined;
+    let callbackArg;
```

Comment:
I think you can just return in the ifs instead of having this vars (not a must, whichever you prefer)